version: "3.8"
services:
  front:
    image: nginx:latest
    ports:
      - "80:80"
    volumes:
      - ./front/nginx:/etc/nginx/conf.d
      - ./front/echoe/dist:/usr/share/nginx/html
  mysql:
    image: mysql:latest
    volumes:
      - ./mysql/data:/var/lib/mysql
      - ./mysql/init:/docker-entrypoint-initdb.d
      - ./mysql/mysql.cnf:/etc/mysql/conf.d/my.cnf
    ports:
      - "3306:3306"
    networks:
      - external_network
    env_file:
      - ./mysql/conf.env
    healthcheck:
      test: [ "CMD", "mysqladmin", "ping", "-h", "localhost", "-u", "root", "-p $$MYSQL_ROOT_PASSWORD" ]
      interval: 5s
      timeout: 5s
      retries: 10
      start_period: 30s

networks:
  internal_network:
    driver: bridge
    internal: true
  external_network:
